### ISSUE TITLE
"Open in Atom" option on changed file is broken

### ISSUE THREAD

**Issue by pmrotule** [desktop/desktop#5762]
## Description

From the list of changed files, you can right-click on a file and choose "Open in Atom" from the context menu. When I do that, it opens a new Atom window then this window crashes. The second time I try to select "Open in Atom", I end up with a error message directly in Github Desktop:

<img width="727" alt="image" src="https://user-images.githubusercontent.com/10983258/46133367-94815700-c240-11e8-8bb9-2b357c0f49bc.png">

I realized afterwards that I can use the option "Open with Default Program" without any problem (my default program is Atom). By the way, I truly miss the previous behavior of Github Desktop 1 which was opening the file with the default program on double click.

## Version

* GitHub Desktop: Version 1.4.1

* Operating system: macOS version 10.12.6 (Macbook Pro 2016)

## Steps to Reproduce

1. Make sure you have Atom installed
2. Change a file from one of the repository managed by Github Desktop
3. Right-click on a changed file and choose the option "Open in Atom"

### Expected Behavior

Switch to Atom and open the changed file (or focus if it was already open).

### Actual Behavior

A new Atom window will open and stay black for a few seconds before it crashes.

**Comment by shiftkey** [comment]
@pmrotule thanks for the report!

That path suggests we queried for Atom's install location while it was in the middle of an update. 

If you run this command from a terminal, does it return anything?

```
mdfind "kMDItemCFBundleIdentifier == 'com.github.atom'"
```

What I believe has happened here is that we've cached the install location for the current session, and it just happened to clash with the time the app was updating - and temporarily moved.

It should be working again on next launch.
